packages/next-app/src/components/OwnedPool.tsx

```diff
@@ -2,23 +2,46 @@ import { Badge, Button, HStack, Text, VStack } from "@chakra-ui/react";
 
 interface IOwnedPool {
   onClick: () => void;
-  address: string;
-  type: string;
-  ratio?: string;
-  name: string;
+  address?: string;
+  poolType?: string | null | undefined;
+  name?: string | null | undefined;
+  id?: string | null | undefined;
+  tokens?:
+    | {
+        symbol: string;
+        weight?: unknown;
+      }[]
+    | null
+    | undefined;
   isSelected: boolean;
 }
 
+function truncateAddress(address: string | undefined) {
+  if (!address) return;
+  const match = address.match(
+    /^(0x[a-zA-Z0-9]{10})[a-zA-Z0-9]+([a-zA-Z0-9]{4})$/
+  );
+  if (!match) return address;
+  return `${match[1]}…${match[2]}`;
+}
+
 export function OwnedPool({
   onClick,
```

Comment:
You could also pass only a `pool` prop from the GraphQL request which would essentially be of the same type of a generated type. IMO it's cleaner to read, and you won't need to redefine a bunch of types that had already been defined by the graphql types.